cylc/flow/task_state.py

```diff
@@ -474,9 +474,11 @@ def _add_prerequisites(self, point, tdef):
             for dependency in dependencies:
                 cpre = dependency.get_prerequisite(point, tdef)
                 if dependency.suicide:
-                    self.suicide_prerequisites.append(cpre)
+                    if cpre not in self.suicide_prerequisites:
```

Comment:
Cool to see the numbers. 

As noted below I had used the set method but ditched it when I backed out of the sorting changes, because I was unsure if it was legit to define `__hash__()` for the mutable Prerequisites type.